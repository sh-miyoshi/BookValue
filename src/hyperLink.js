import React, { Component } from 'react';
import { Text, Linking } from 'react-native';

export class HyperLink extends Component {
  constructor(props) {
    super(props);
    this._goToURL = this._goToURL.bind(this);
  }

  render() {
    return (
      <Text style={styles.linkStyle} onPress={this._goToURL}>
        {this.props.title}
      </Text>
    );
  }

  _goToURL() {
    Linking.canOpenURL(this.props.url).then(supported => {
      if (supported) {
        Linking.openURL(this.props.url);
      } else {
        console.log('Don\'t know how to open URI: ' + this.props.url);
      }
    });
  }
}

// CSS
const styles = {
  linkStyle: {
    marginTop: 30,
    fontSize: 10,
    textDecorationLine: 'underline'
  },
}