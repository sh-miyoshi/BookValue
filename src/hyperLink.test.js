import React from 'react'
import { HyperLink } from './hyperLink'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer
    .create(<HyperLink url="https://www.google.com/" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
})