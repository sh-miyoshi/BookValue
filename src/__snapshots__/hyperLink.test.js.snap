// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  onPress={[Function]}
  style={
    Object {
      "fontSize": 10,
      "marginTop": 30,
      "textDecorationLine": "underline",
    }
  }
/>
`;
